addons/hr_timesheet/static/tests/hr_timesheet_common_tests.js

```diff
@@ -0,0 +1,136 @@
+                task_id: { string: "Task", type: "many2one", relation: "project.task" },
+                unit_amount: { string: "Unit Amount", type: "integer" },
+            },
+            records: [
+                { id: 1, project_id: 1, task_id: 3, unit_amount: 1 },
+                { id: 2, project_id: 1, task_id: false, unit_amount: 1 },
+                { id: 3, project_id: false, task_id: false, unit_amount: 1 },
+            ],
+        },
+        'project.project': {
+            fields: {
+                name: { string: "Name", type: "string" },
+            },
+            records: [
+                { id: 1, name: "Project 1" },
+            ],
+        },
+        'project.task': {
+            fields: {
+                name: { string: "Name", type: "string" },
+                project_id: { string: "Project", type: "many2one", relation: "project.project" },
+            },
+            records: [
+                { id: 1, name: "Task 1\u00A0AdditionalInfo", project_id: 1 },
+                { id: 2, name: "Task 2\u00A0AdditionalInfo", project_id: 1 },
+                { id: 3, name: "Task 3\u00A0AdditionalInfo", project_id: 1 },
+            ],
+        }
+    },
+    views: {
+        "account.analytic.line,false,form": `
+            <form>
+                <field name="project_id"/>
+                <field name="task_id"/>
+                <field name="unit_amount"/>
+            </form>
+        `,
+        "account.analytic.line,false,list": `
+            <tree editable="bottom">
+                <field name="project_id"/>
+                <field name="task_id"/>
+                <field name="unit_amount"/>
+            </tree>
+        `,
+    },
+}));
+
+export function updateArch(serverData, fieldNameWidgetNameMapping = {}, fieldNameContextMapping = {}) {
+    for (const viewKey in serverData.views) {
+        for (const [fieldName, widgetName] of Object.entries(fieldNameWidgetNameMapping)) {
+            serverData.views[viewKey] = serverData.views[viewKey].replace(
+                `name="${fieldName}"`,
+                `name="${fieldName}" widget="${widgetName}"`
+            );
+        }
+        for (const [fieldName, context] of Object.entries(fieldNameContextMapping)) {
+            serverData.views[viewKey] = serverData.views[viewKey].replace(
+                `name="${fieldName}"`,
+                `name="${fieldName}" context="${context}"`
+            );
+        }
+    }
+}
+
+export function addFieldsInArch(serverData, fields = [], beforeField) {
+    let fieldsArch = "";
+    for (const field of fields) {
+        fieldsArch += `<field name="${field}"/>
+                `;
+    }
+    for (const viewKey in serverData.views) {
+        serverData.views[viewKey] = serverData.views[viewKey].replace(
+            `<field name="${beforeField}"`,
+            `${fieldsArch}<field name="${beforeField}"`
+        );
+    }
+}
+
+function patchSessionWithCleanup() {
```

Comment:
No need to export this function? 🙂 🤔

Reply:
I moved the code directly in `setupTestEnv `